web/html/src/components/states-picker.tsx

```diff
@@ -0,0 +1,447 @@
+import * as React from "react";
+
+import _partition from "lodash/partition";
+import _sortBy from "lodash/sortBy";
+import _unionBy from "lodash/unionBy";
+
+import { SectionToolbar } from "components/section-toolbar/section-toolbar";
+
+import { DEPRECATED_unsafeEquals } from "utils/legacy";
+
+import { AsyncButton } from "../components/buttons";
+import { TextField } from "../components/fields";
+import { Messages, MessageType } from "../components/messages";
+import { Utils as MessagesUtils } from "../components/messages";
+import { RankingTable } from "../components/ranking-table";
+import { SaltStatePopup } from "../components/salt-state-popup";
+import Network from "../utils/network";
+
+function channelKey(channel) {
+  return channel.label;
+}
+
+function channelIcon(channel) {
+  let iconClass, iconTitle, iconStyle;
+  if (channel.type === "state") {
+    iconClass = "fa spacewalk-icon-salt-add";
+    iconTitle = t("State Configuration Channel");
+  } else if (channel.type === "internal_state") {
+    iconClass = "fa spacewalk-icon-salt-add";
+    iconTitle = t("Internal State");
+    iconStyle = { border: "1px solid black" };
+  } else {
+    iconClass = "fa spacewalk-icon-software-channels";
+    iconTitle = t("Normal Configuration Channel");
+  }
+
+  return <i className={iconClass} title={iconTitle} style={iconStyle} />;
+}
+
+type StatesPickerProps = {
+  type?: string | any;
```

Comment:
`string | any` collapses into `any`, is there a way we can get rid of `any` here?